node/lib/Munin/Node/SpoolReader.pm

```diff
@@ -96,22 +104,18 @@ sub list
 }
 
 
+# The third argument ($submission_function) is optional: it is used for processing partial results,
+# in order to avoid slurping the full data content into RAM.  The result is empty, if this function
+# is given.
 sub _cat_multigraph_file
 {
-    my ($self, $service, $timestamp, $max_samples_per_service) = @_;
-
-    # Default $max_samples_per_service is 5, in order to have a 5x time
-    # increase in catchup.  This enables to not overwhelm the munin-update when
-    # there is big backlog to handle. Use "0" to send an infinite number of
-    # samples
-    $max_samples_per_service = 5 if (! defined $max_samples_per_service);
+    my ($self, $service, $timestamp, $submission_function) = @_;
 
     my $data = "";
 
     rewinddir $self->{spooldirhandle}
         or die "Unable to reset the spool directory handle: $!";
 
-    my $nb_samples_sent = 0;
     foreach my $file (readdir $self->{spooldirhandle}) {
```

Comment:
change the loop for readdir, to sort them (since we want older samples first), by adding `sort { $a cmp $b }` before readdir:
```
foreach my $file (sort { $a cmp $b } readdir $handler) {
```
Then we get incremental correctly. Tested with:
```
#!/usr/bin/perl


use IO::File;

print "reading dir\n";
opendir $handler, '/var/lib/munin-spool';
my $diff=0, $ago=0;
foreach my $file (sort { $a cmp $b } readdir $handler) {
        $file =~ m/^munin-daemon\.([a-z_-]+)\.(\d+)\.(\d+)$/;
        next unless "$1" eq 'nfs_client';
        $diff=$2-$ago;
        $ago=$2;
        print "service $1: $2 epoch, diff=$diff, sample_size: $3:";
        print "$file\n";
}
```

Reply:
> foreach my $file (sort { $a cmp $b } readdir $handler) {

Excuse me, but my perl-foo is very rusty:
Can we remove the `{$a cmp $b}` from above, since it is the default behaviour?

Reply:
If that is the case, I suppose why not. I just found the solution on the internet, but if shorter is possible, then sure!
Also tested it, and the result was the same!

Reply:
Thanks for testing!